cogs/get_token_authorisation.py

```diff
@@ -60,17 +58,21 @@ async def get_token_authorisation(self, ctx: TeXBotApplicationContext) -> None:
         async with http_session, http_session.get(REQUEST_URL) as http_response:
             response_html: str = await http_response.text()
 
-        parsed_html: bs4.Tag | bs4.NavigableString | None = BeautifulSoup(
+        page_title: bs4.Tag | bs4.NavigableString | None = BeautifulSoup(
             response_html,
             "html.parser",
-        ).find("ul", {"id": "ulOrgs"})
+        ).find("title")
+
+        if not page_title:
+            await self.command_send_error(ctx, message="Something has gone very wrong.")
+            return
 
-        if parsed_html is None or isinstance(parsed_html, bs4.NavigableString):
-            logger.debug(
-                "No admin table was found, meaning the token provided "
-                "does not have admin access to any societies.",
+        if isinstance(page_title, bs4.NavigableString) and "Login" in page_title or "Login" in page_title.string:  # type: ignore[operator, union-attr]  # noqa: E501
```

Comment:
Flip guard clause check & improve invalid type handling rather than ignoring errors.

Fix:
```python
BAD_TOKEN_MESSAGE: Final[str] = (
    "Unable to fetch profile page because the token was not valid."
)
profile_page_invalid_message: str | None = None

if isinstance(page_title, bs4.Tag):
    if not hasattr(page_title, "string"):
        profile_page_invalid_message = "Profile page had invalid format"

    else:
        title_string: object = getattr(page_title, "string")
        if not isinstance(title_string, str):
            raise TypeError

        if "login" in title_string.lower():
            profile_page_invalid_message = BAD_TOKEN_MESSAGE

elif isinstance(page_title, bs4.NavigableString):
    if "login" in page_title.lower():
        profile_page_invalid_message = BAD_TOKEN_MESSAGE

else:
    raise TypeError

if profile_page_invalid_message is not None:
    logger.warning(profile_page_invalid_message)
    await ctx.respond(content=profile_page_invalid_message)
    return
```

(This could be improved by using Python's new match-case statement, but that's not too big of an issue)